### PR TITLE
Add Jenkinsfile

### PR DESCRIPTION
**Pull request by mikenairn**
## Motivation

Enable building the app via the Mobile CI/CD service.

## Description

Adds a Jenkinsfile

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+//CODE_SIGN_PROFILE_ID = "redhat-dist-dp"
+//BUILD_CONFIG = "Debug"
+
+// sample values commented below are for https://github.com/feedhenry-templates/helloworld-ios
+/* ------------- use these to hardcode values in Jenkinsfile ---------------- */
+PROJECT_NAME = "helloworld-ios-app"
+INFO_PLIST = "helloworld-ios-app/helloworld-ios-app-Info.plist"
+VERSION = "1.0.0"
+SHORT_VERSION = "1.0"
+BUNDLE_ID = "com.feedhenry.helloworld-ios-app"
```
Comment (wei-lee) [inline Jenkinsfile]: I think those values need to be checked again. They are not matching this project.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+node('ios') {
+    stage('Checkout') {
+        checkout scm
+    }
+
+    stage('Prepare') {
+        if(params.FH_CONFIG_CONTENT) {
+            writeFile file: "${PROJECT_NAME}/fhconfig.plist", text: FH_CONFIG_CONTENT
+        }
```
Comment (wei-lee) [inline Jenkinsfile]: this is not required for this app.

File: Jenkinsfile (hunk 1 above)
Comment (odra) [reply Jenkinsfile]: we can use: 

```
INFO_PLIST = 'ios-showcase-template/AppConfig.plist'
VERSION = "1.0.0"
SHORT_VERSION = "1.0"
BUNDLE_ID = 'org.aerogear.ios-showcase-template'
```

Comment (mikenairn) [reply Jenkinsfile]: yep, my bad, will update.

File: Jenkinsfile (hunk 2 above)
Comment (mikenairn) [reply Jenkinsfile]: It is optional, but i guess since we never plan on using these with 4x build farm we can remove it.

File: Podfile [hunk 3]
```diff
@@ -20,7 +20,7 @@ target 'ios-showcase-template' do
   pod 'SwiftKeychainWrapper'
   pod 'NotificationBannerSwift'
   pod 'RealmSwift'
-  pod 'SCLAlertView', :git => 'https://github.com/vikmeup/SCLAlertView-Swift.git', :branch => 'master'
+  pod 'SCLAlertView', :git => 'https://github.com/vikmeup/SCLAlertView-Swift.git', :branch => 'swift-2.3'
```
Comment (mikenairn) [inline Podfile]: @wei-lee @wtrocki We were unable to get the app to build via the Mobile CI/CD service without changing this. Do you know if we need to use master here or can we use this branch? Are there updates we need to make to the Mobile CI/CD service to allow the master branch to work? It is currently throwing this error:

```
bash-3.2$ pod install                                               
Analyzing dependencies                                              
Pre-downloading: `SCLAlertView` from `https://github.com/vikmeup/SCLAlertView-Swift.git`, branch `master`                               
[!] Unable to find a specification for 'SCLAlertView'.              

[!] Unable to load a podspec from `SCLAlertView.podspec`, skipping: 

Pod::DSLError
```

Comment (odra) [reply Podfile]: I tried to run `pod install` from my mac and got the same error

**Comment by wtrocki** [comment]
@mikenairn This looks like technical debt that was moved from feedhenry repo. 
This library is used almost everywhere. Locking that to latest release it's not going to fix it so your change seems to be quick patch. Author recently changed master and it breaks. 

btw. Any reason why we do not use fastline? 
xcodebuilds are really hard way to get things done. 
Our current builds in circleci using fastlane.

**Comment by odra** [comment]
No particular reason, we just went with what we already had

**Comment by wtrocki** [comment]
@odra Sorry. I was under impression that you are building this from scratch. It's great so! However we also need to make sure that circle Ci build will pass here.

**Comment by odra** [comment]
but I agree - we could be using fastlane in the jenkinsfile

**Review by odra — APPROVED** [review]

**Comment by odra** [comment]
The build works after updating cocoapods to 1.5.0

**Comment by wtrocki** [comment]
@mikenairn Do you also target to allow builds for PR's with some github integration? or this will be only self hosted, on demand build? It will be cool to get IPA/APK file on demand :) 
I'm just curious.

**Comment by mikenairn** [comment]
@wtrocki Purpose of this was to just get the Jenkinsfiles in the repos so that we could build the apps using the Mobile CI/CD service. 

That said, it would probably be a good idea to have some way of triggering a build of a PR branch against a running Mobile CI/CD service for verification of the build and to easily create an IPA/APK to test on device.

**Comment by wtrocki** [comment]
I reviewed the script and happy to get that merged. Have not tested that.

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    BUILD_CONFIG = "Release"
+}
+
+node('ios') {
+    stage('Checkout') {
+        checkout scm
+    }
+
+    stage('Prepare') {
+        sh 'pod install'
```
Comment (wtrocki) [inline Jenkinsfile:34]: [Question] How do we refresh podspecs cache on this machine.

**Review by wtrocki — APPROVED** [review]

File: Jenkinsfile (hunk 4 above)
Comment (odra) [reply Jenkinsfile:34]: we have a cronjob that refreshes it every 3 hours

**Pull request merged**